### DISCUSSION TITLE
#Image Size

### DISCUSSION THREAD

**Discussion by NigussieAbate** [mrdbourke/pytorch-deep-learning#1006] [Q&A]
Can we apply ViT for datasets with image height and width are not the same? E.g. 288 x 640

**Comment by LuluW8071** [comment]
Yes you can as long as all the images have same resolution otherwise you may need to resize it or use custom padding/cropping to avoid shape mismatch errors.

**Comment by aniruddha76** [comment]
Yes, you can use ViT on images with different height and width, like 288 x 640. ViTs typically divide images into smaller patches, so as long as the image can be divided into patches (like 16x16 or 32x32), it should work. If the image size isn't an exact multiple of the patch size, you might need to resize or pad the image to fit the patches perfectly.

mark this as answer if you find this useful
